frontend/src/pages/WorkspaceLocked/WorkspaceLocked.tsx

```diff
@@ -0,0 +1,84 @@
+/* eslint-disable react/no-unescaped-entities */
+import './WorkspaceLocked.styles.scss';
+
+import { CreditCardOutlined, LockOutlined } from '@ant-design/icons';
+import { Button, Card, Typography } from 'antd';
+import updateCreditCardApi from 'api/billing/checkout';
+import { getFormattedDate } from 'container/BillingContainer/BillingContainer';
+import useLicense from 'hooks/useLicense';
+import { useCallback, useEffect, useState } from 'react';
+import { useMutation } from 'react-query';
+import { useSelector } from 'react-redux';
+import { AppState } from 'store/reducers';
+import { License } from 'types/api/licenses/def';
+import AppReducer from 'types/reducer/app';
+
+export default function WorkspaceBlocked(): JSX.Element {
+	const { role } = useSelector<AppState, AppReducer>((state) => state.app);
+	const isAdmin = role === 'ADMIN';
+	const [activeLicense, setActiveLicense] = useState<License | null>(null);
+
+	const { isFetching, data: licensesData } = useLicense();
+
+	useEffect(() => {
+		const activeValidLicense =
+			licensesData?.payload?.licenses?.find(
+				(license) => license.isCurrent === true,
+			) || null;
+
+		setActiveLicense(activeValidLicense);
+	}, [isFetching, licensesData]);
+
+	const { mutate: updateCreditCard, isLoading } = useMutation(
+		updateCreditCardApi,
+		{
+			onSuccess: (data) => {
+				window.open(data.payload?.redirectURL);
```

Comment:
This was opening up in pop-up and blocked by the browser. Is there a way to avoid it? Or shall I send the redirect from the server itself?

Reply:
Addressed